src/usePopper.js

```diff
@@ -98,35 +98,32 @@ export const usePopper = (
   ]);
 
   const popperInstanceRef = React.useRef();
-  const createPopper = React.useMemo(
-    () => options.createPopper || defaultCreatePopper,
-    [options.createPopper]
-  );
 
   useIsomorphicLayoutEffect(() => {
-    let popperInstance = null;
-    if (referenceElement != null && popperElement != null) {
-      popperInstance = createPopper(
-        referenceElement,
-        popperElement,
-        popperOptions
-      );
-
-      popperInstanceRef.current = popperInstance;
+    if (popperInstanceRef.current) {
+      popperInstanceRef.current.setOptions(popperOptions);
+    }
+  }, [popperOptions]);
+
+  useIsomorphicLayoutEffect(() => {
+    if (referenceElement == null || popperElement == null) {
+      return;
     }
 
+    const createPopper = options.createPopper || defaultCreatePopper;
+    const popperInstance = createPopper(
+      referenceElement,
+      popperElement,
+      popperOptions
+    );
+
+    popperInstanceRef.current = popperInstance;
+
     return () => {
-      popperInstance != null && popperInstance.destroy();
+      popperInstance.destroy();
       popperInstanceRef.current = null;
     };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
`react-hooks/exhaustive-deps` unfortunately doesn't/cannot handle checking the dependency arrays for "custom" hooks.
Do we really have to avoid using `useLayoutEffect` in SSR env?

Reply:
Without, SSR customers would get a warning that would be very difficult to get rid of, so I don't think there are better alternatives.